### PR TITLE
1.19 Release Calendar (alpha version)

### PR DESCRIPTION
**Pull request by onlydole**
Welcome to the Kubernetes 1.19 release calendar!

I am jumping on this a bit early so we can discuss it as a release team before and during the first week of our schedule.

/kind documentation
/hold for discussion

### CODE REVIEW

**Comment by justaugustus** [comment]
Explicit hold for Branch Manager schedule reshuffles:
/hold

File: releases/release-1.19/README.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+- **Tuesday, May 5**: Week 4 - [Enhancements Freeze]
+- **Thursday, June 11**: Week 9 - [Code Freeze]
+- **Tuesday, June 23**: Week 11 - Docs must be completed and reviewed
+- **Tuesday, June 30**: Week 12 - Kubernetes v1.19.0 released
+
+
+## Timeline
+
+
+| **What** | **Who** | **Apr** | **May** | **Jun** | **Jul** | **WEEK** | **CI SIGNAL** |
```
Comment (justaugustus) [inline releases/release-1.19/README.md:40]: @onlydole -- Here's a rough draft of what I'm thinking about on the Branch Management side:


| **What** | **Who** | **Apr** | **May** | **Jun** | **Jul** | **WEEK** | **CI SIGNAL** |
| --- | --- | --- | --- | --- | --- | --- | --- |
| Start of Release Cycle | Lead | Mon 13 | | | | week 1 | [master-blocking] |
| 1.19.0-alpha.2 released | Branch Manager | Tue 21 | | | | week 2 | |
| **Begin [Enhancements Freeze]** (EOD PST) | Enhancements Lead | | Tues 5 | | | week 4 | [master-blocking], [master-informing] |
| 1.19.0-alpha.3 released | Branch Manager | | Tue 5 | | | week 4 | |
| 1.19.0-beta.0 released | Branch Manager | | Tue 19 | | | week 6 | |
| 1.19.0-beta.1 released | Branch Manager | | Tue 26 | | | week 7 | |
| 1.19.0-beta.2 released | Branch Manager | | | Tue 2 | | week 8 | |
| **Begin [Code Freeze]** (EOD PST) | Branch Manager | | | Thu 11 | | week 9 | |
| 1.19.0-rc.1 released | Branch Manager | | | Thu 11 | | week 9 | |
| release-1.19 branch created | Branch Manager | | | Thu 11 | | week 9 | |
| release-1.19 jobs created | Branch Manager | | | Thu 11 | | week 9 | |
| release-1.15 jobs removed | Branch Manager | | | Thu 11 | | week 9 | |
| 1.19.0-rc.2 released | Branch Manager | | | Tue 16 | | week 10 | |
| **Cherry Pick Deadline** (EOD PST) | Branch Manager | | | Tue 23 | | week 11 | |
| 1.19.0-rc.3 released | Branch Manager | | | Thu 25 | | week 11 | |
| **v1.19.0 released** | Branch Manager | | | Tue 30 | | week 12 | |

Consider this alpha as well. Just putting it up so we can start discussing removing fast-forwards (ref: https://github.com/kubernetes/release/issues/1207).

tl;dr:
- cut alphas every two weeks, then:
- cut 3 betas weekly on Tuesdays, then:
- start Code Freeze, which would now be signified by cutting RC1 and the release-1.19 branch
- cherry pick deadline happens on a week after RC2 and BEFORE RC3
- RC3 is released
- 1.19.0 is released (and likely has the same content as RC3)

By doing the branch cut at Code Freeze, that means:
- `master` immediately opens for Kubernetes 1.20 and Code Thaw is obsolete
- any content that needs to land in 1.19 needs to be cherry picked

@kubernetes/release-engineering -- Thoughts on this?

Comment (tpepper) [reply releases/release-1.19/README.md:40]: Feels sensible to me.

Comment (tpepper) [reply releases/release-1.19/README.md:40]: I think the Cherry Pick deadline could still be Thu 25.

Comment (alejandrox1) [inline releases/release-1.19/README.md:40]: It has been mentioned in the past that
```
| **What** | **Who** | **When* | **WEEK** | **CI SIGNAL** |
```
is better than
```
| **What** | **Who** | **Apr** | **May** | **Jun** | **Jul** | **WEEK** | **CI SIGNAL** |
```

specially for mobile

File: releases/release-1.19/README.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
+# Kubernetes 1.19
+
+
+#### Links
+
+* [This document](https://git.k8s.io/sig-release/releases/release-1.19/README.md)
+* [Release Team](https://git.k8s.io/sig-release/releases/release-1.19/release_team.md)
+* [Meeting Minutes](http://bit.ly/k8s119-releasemtg) (join [kubernetes-sig-release@] to receive meeting invites)
```
Comment (alejandrox1) [inline releases/release-1.19/README.md:8]: http://bit.ly/k8s119-releasemtg doesn't work, same for retro, enhancements doc, etc.

File: releases/release-1.19/README.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+| Start of Release Cycle | Lead | Mon 13 | | | | week 1 | [master-blocking] |
+| Start Enhancements Tracking | Enhancements Lead | Tue 14 | | | | week 1 | |
+| Schedule finalized | Lead | Fri 17 | | | | week 1 | |
+| Team finalized | Lead | Fri 17 | | | | week 1 | |
+| 1.19.0-alpha.1 released | Branch Manager | Tue 21 | | | | week 2 | |
+| Start Release Notes Draft | Release Notes Lead | Tue 28 | | | | week 3 | |
+| **Begin [Enhancements Freeze]** (EOD PST) | Enhancements Lead | | Tues 5 | | | week 4 | [master-blocking], [master-informing] |
+| 1.19.0-alpha.2 released | Branch Manager | | Tue 12 | | | week 5 | |
+| 1.19.0-alpha.3 released | Branch Manager | | Tue 19 | | | week 6 | |
+| release-1.14 jobs removed | Branch Manager | | Tue 26 | | | week 7 | |
```
Comment (alejandrox1) [inline releases/release-1.19/README.md:51]:
```suggestion
| release-1.15 jobs removed | Branch Manager | | Tue 26 | | | week 7 | |
```

File: releases/release-1.19/README.md (hunk 1 above)
Comment (onlydole) [reply releases/release-1.19/README.md:40]: Ohhhh - I like this much better, too (and it'll help the next release lead)

**Comment by onlydole** [comment]
I like all the comments so far! I'll address feedback later today. I'll keep this PR "eventually consistent" with the nits and feedback. 😄

File: releases/release-1.19/README.md (hunk 2 above)
Comment (palnabarun) [reply releases/release-1.19/README.md:8]: @alejandrox1 I created the short link for the 1.19 Enhancements tracking sheet. The link should work now.

cc: @onlydole

Comment (palnabarun) [reply releases/release-1.19/README.md:8]: I will work on the permissions bit as the team gets stabilized.

Comment (onlydole) [reply releases/release-1.19/README.md:8]: Thank you, @palnabarun!

File: releases/release-1.19/README.md (hunk 1 above)
Comment (onlydole) [reply releases/release-1.19/README.md]: @justaugustus and @tpepper - I have updated the schedule to reflect these changes, let me know if you like how that looks now.

File: releases/release-1.19/README.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+- **Thursday, June 11**: Week 9 - [Code Freeze]
+- **Tuesday, June 23**: Week 11 - Docs must be completed and reviewed
+- **Tuesday, June 30**: Week 12 - Kubernetes v1.19.0 released
+
+
+## Timeline
+
+
+| **What** | **Who** | **When** | **WEEK** | **CI SIGNAL** |
+| --- | --- | --- | --- | --- | --- | --- | --- |
```
Comment (justaugustus) [inline releases/release-1.19/README.md:41]:
```suggestion
|---|---|---|---|---|
```

File: releases/release-1.19/README.md (hunk 1 above)
Comment (saschagrunert) [reply releases/release-1.19/README.md]: Sounds good to me! :) :+1:

Comment (cpanato) [reply releases/release-1.19/README.md]: sgtm as well, thanks!

Comment (palnabarun) [reply releases/release-1.19/README.md]: LGTM from my side too. Thank you! :)

File: releases/release-1.19/README.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+| release-1.19 jobs created | Branch Manager | Thu, June 11 | week 9 | |
+| release-1.15 jobs removed | Branch Manager | Thu, June 11 | week 9 | |
+| Burndown Meetings daily| Lead | Mon, June 15 | week 10 | |
+| Docs deadline - PRs ready for review | Docs Lead | Mon, June 15 | week 10 | |
+| 1.19.0-rc.2 released | Branch Manager | Tue, June 16 | week 10 | |
+| Docs complete - All PRs reviewed and ready to merge | Docs Lead | Mon, June 22 | week 11 | |
+| 1.19.0-rc.3 released | Branch Manager | Tue, June 23 | week 11 | |
+| **Cherry Pick Deadline** (EOD PST) | Branch Manager | Thu, June 25 | week 11 | |
+| **v1.19.0 released** | Branch Manager | Tue, June 30 | week 12 | |
+| Release retrospective | Community | Thu, July 2 | week 12 | |
```
Comment (alejandrox1) [inline releases/release-1.19/README.md:69]: nit: may possibly want to schedule retro for week after release (?) - in case delays happen and to give you room for all interviews and media things

File: releases/release-1.19/README.md (hunk 1 above)
Comment (mkorbi) [reply releases/release-1.19/README.md]: looks good! lgtm 👍

File: releases/release-1.19/README.md (hunk 5 above)
Comment (jeremyrickard) [reply releases/release-1.19/README.md:69]: +1 to @alejandrox1 ’s comment

Comment (mrbobbytables) [reply releases/release-1.19/README.md:69]: Another +1 here as well.

Comment (justaugustus) [reply releases/release-1.19/README.md:69]: +1 too :)

**Comment by onlydole** [comment]
@justaugustus and @tpepper - that should do the trick! Any final nits or crits before we approve?

**Comment by justaugustus** [comment]
I think I was the only one with a nit, so we should be good to go!
/lgtm
/approve
/hold cancel
